### PR TITLE
Add test case for Content Based Routing

### PR DESCRIPTION
**Pull request by Maninda**
## Purpose
Add test case for Content Based Routing: Resolves https://github.com/wso2/ballerina-integrator/issues/41

### CODE REVIEW

File: examples/integration-tutorials/routing-requests-based-on-message-content/test/content_based_routing_test.bal [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            var resPayload = response.getJsonPayload();
+            if (resPayload is json) {
+                var payloadJson = response.getJsonPayload();
+                if(payloadJson is json) {
+                    test:assertEquals(payloadJson["confirmed"], expected["confirmed"], msg = "Response mismatch!");
+                }
+            } else {
+                test:assertFail(msg = "Payload from reservation service is invalid");
+            }
+        }
```
Comment (Asitha) [inline examples/integration-tutorials/routing-requests-based-on-message-content/test/content_based_routing_test.bal]: Don't we need to fail on else block here?

File: examples/integration-tutorials/routing-requests-based-on-message-content/test/content_based_routing_test.bal [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    if (response is http:Response) {
+        test:assertEquals(response.statusCode, 200, msg = "Reserve-Appointment service did not respond with 200 OK signal!");
+        string|error receivedPayload = response.getTextPayload();
+        if(receivedPayload is string) {
+            var resPayload = response.getJsonPayload();
+            if (resPayload is json) {
+                var payloadJson = response.getJsonPayload();
+                if(payloadJson is json) {
+                    test:assertEquals(payloadJson["confirmed"], expected["confirmed"], msg = "Response mismatch!");
+                }
```
Comment (Asitha) [inline examples/integration-tutorials/routing-requests-based-on-message-content/test/content_based_routing_test.bal]: Don't we need to fail on else block?

File: examples/integration-tutorials/routing-requests-based-on-message-content/test/content_based_routing_test.bal (hunk 1 above)
Comment (Maninda) [reply examples/integration-tutorials/routing-requests-based-on-message-content/test/content_based_routing_test.bal]: Removed unwanted JSON conversion

File: examples/integration-tutorials/routing-requests-based-on-message-content/test/content_based_routing_test.bal (hunk 2 above)
Comment (Maninda) [reply examples/integration-tutorials/routing-requests-based-on-message-content/test/content_based_routing_test.bal]: Added the else case test failure

**Pull request merged**